core/src/trezor/ui/layouts/common.py

```diff
@@ -26,21 +26,20 @@ async def button_request(
         log.debug(__name__, "ButtonRequest.type=%s", br_type)
     workflow.close_others()
     if pages is not None:
-        await ctx.call(ButtonRequest(code=code, pages=pages), ButtonAck)
+        await context.maybe_call(ButtonRequest(code=code, pages=pages), ButtonAck)
     else:
-        await ctx.call(ButtonRequest(code=code), ButtonAck)
+        await context.maybe_call(ButtonRequest(code=code), ButtonAck)
 
 
 async def interact(
-    ctx: wire.GenericContext,
     layout: LayoutType,
     br_type: str,
     br_code: ButtonRequestType = ButtonRequestType.Other,
 ) -> Any:
     if hasattr(layout, "page_count") and layout.page_count() > 1:  # type: ignore [Cannot access member "page_count" for type "LayoutType"]
         # We know for certain how many pages the layout will have
-        await button_request(ctx, br_type, br_code, pages=layout.page_count())  # type: ignore [Cannot access member "page_count" for type "LayoutType"]
-        return await ctx.wait(layout)
+        await button_request(br_type, br_code, pages=layout.page_count())  # type: ignore [Cannot access member "page_count" for type "LayoutType"]
+        return await context.wait(layout)
     else:
```

Comment:
The code we call is almost the same in both if-else branches, we might simplify it a little, e.g.:

```python
if hasattr(layout, "page_count") and layout.page_count() > 1:
    page_count = layout.page_count()
else:
    page_count = None
await button_request(br_type, br_code, pages=page_count)
return await context.wait(layout)
```

Reply:
81305badef3cbadaa7de0b696ce83953dae4089f